httpbakery/client.go

```diff
@@ -343,14 +344,47 @@ func (c *Client) setRequestBody(req *http.Request, body io.ReadSeeker) error {
 	if body == nil {
 		return nil
 	}
-	if req.Body == nil {
-		req.Body = ioutil.NopCloser(body)
-	} else {
-		_, err := body.Seek(0, 0)
-		if err != nil {
+	if req.Body != nil {
+		req.Body.Close()
```

Comment:
We know statically that it is a readStopper (an earlier version of the code actually did req.Body.(*readStopper).stop()) and therefore cannot return an error. Even if it wasn't, there's nothing useful that the error can tell us from a Reader - it's not like we're checking whether previously buffered writes can fail. If the Close fails, so what? FWIW the net/http package also ignores the error from Request.Body.Close.